shared/src/commonMain/kotlin/com/example/tasky/model/ResultWrapper.kt

```diff
@@ -0,0 +1,11 @@
+package com.example.tasky.model
+
+sealed interface ResultWrapper<out D> {
+    data class Success<out D>(
+        val data: D,
+    ) : ResultWrapper<D>
+
+    data class Error<out D>(
+        val error: ErrorResponse,
+    ) : ResultWrapper<D>
+}
```

Comment:
It seems like your `ResultWrapper` is a bit inflexible since you already have to decide about the exact error string where you create it. If you'd like to support localization, this is difficult depending on where this result is created. I'm personally using this variant which I found to be really useful over time. It works with extendable enums rather than hardcoded error strings. These enums can then be mapped to actual error messages in the UI:
```
sealed interface Result<out D, out E: Error> {
    data class Success<out D>(val data: D): Result<D, Nothing>
    data class Error<out E: com.plcoding.tasky.core.domain.util.Error>(val error: E): Result<Nothing, E>
}

inline fun <T, E: Error, R> Result<T, E>.map(map: (T) -> R): Result<R, E> {
    return when(this) {
        is Result.Error -> Result.Error(error)
        is Result.Success -> Result.Success(map(data))
    }
}

fun <T, E: Error> Result<T, E>.asEmptyDataResult(): EmptyResult<E> {
    return map {  }
}

inline fun <T, E: Error> Result<T, E>.onSuccess(action: (T) -> Unit): Result<T, E> {
    return when(this) {
        is Result.Error -> this
        is Result.Success -> {
            action(data)
            this
        }
    }
}
inline fun <T, E: Error> Result<T, E>.onError(action: (E) -> Unit): Result<T, E> {
    return when(this) {
        is Result.Error -> {
            action(error)
            this
        }
        is Result.Success -> this
    }
}

typealias EmptyResult<E> = Result<Unit, E>
```
It handles data and errors in a type-safe way. You'll need a marker interface to mark error enums:
```
interface Error
```
And then you can extend this across the codebase with different error types, e.g.:
```
sealed interface DataError: Error {
    enum class Remote: DataError {
        NOT_FOUND,
        NO_INTERNET,
        UNAUTHORIZED,
        ...
    }
    
    enum class Local: DataError {
        DISK_FULL,
        ...
    }
}
```
Works great:
```
repository.signUp(payload).onSuccess { responseData ->

}.onError { error -> 
    _state.update { it.copy(error = error) }
}
```